pkg/server/cluster_server.go

```diff
@@ -45,16 +62,70 @@ type clusterServer struct {
 func NewClusterServer(kubeConfig, apiserverURL string) (Server, error) {
 	restConfig, err := getClientConfig(kubeConfig)
 	if err != nil {
-		return nil, fmt.Errorf("Failed to create Kubernetes rest client: %v", err)
+		return nil, errors.Wrapf(err, "Failed to create Kubernetes rest client")
 	}
 
-	mc := v1.NewForConfigOrDie(restConfig)
+	mc, err := v1.NewForConfig(restConfig)
+	if err != nil {
+		return nil, errors.Wrapf(err, "creating mc client")
+	}
+	oc, err := oconfigv1.NewForConfig(restConfig)
+	if err != nil {
+		return nil, errors.Wrapf(err, "creating config client")
+	}
 	return &clusterServer{
 		machineClient:  mc,
+		configClient:   *oc,
 		kubeconfigFunc: func() ([]byte, []byte, error) { return kubeconfigFromSecret(bootstrapTokenDir, apiserverURL) },
 	}, nil
 }
 
+// remoteAddrIsProvisioned returns true if the given IP address responds on a known port.
+// If that's true, the machine is already up and should not be served Ignition.
+func remoteAddrIsProvisioned(remoteAddr string) (string, error) {
+	var wg sync.WaitGroup
+	var provisionedPort string
+	wg.Add(len(machineProvisionedPorts))
+	for _, port := range machineProvisionedPorts {
+		go func(port string) {
```

Comment:
can this variable be named something else? avoid shadowing and the errors that come with it

Reply:
Might as well shadow - [it'd be a trap](https://github.com/golang/go/wiki/CommonMistakes#using-goroutines-on-loop-iterator-variables) to use the outer loop variable right?

Reply:
I don't think it should matter if the function arg has a different variable name so long as the range value is passed to it properly. That linked example tho 😳 . Guess it doesn't matter too much if there have't been complaints about it?
```go
	for _, p := range machineProvisionedPorts {
 		go func(port string) {
			// use `port` here
 		}(p)
	}
```